src/global/reco/ParticleFlow_factory.cc

```diff
@@ -0,0 +1,126 @@
+#include <JANA/JException.h>
+// event data model definitions
+#include <edm4eic/ClusterCollection.h>
+#include <edm4eic/ReconstructedParticleCollection.h>
+// for geometry services
+#include <DD4hep/Detector.h>
+// factory-specific includes
+#include "ParticleFlow_factory.h"
+
+namespace eicrecon {
+
+  void ParticleFlow_factory::Init() {
+
+    auto app = GetApplication();
+
+    // This prefix will be used for parameters
+    std::string plugin_name = GetPluginName();
+    std::string param_prefix = plugin_name + ":" + GetTag();
+
+    // SpdlogMixin logger initialization, sets m_log
+    InitLogger(GetApplication(), GetPrefix(), "info");
+
+    // Algorithm configuration
+    auto cfg = GetDefaultConfig();
+
+    app->SetDefaultParameter(param_prefix + ":flowAlgo",      cfg.flowAlgo);
+    app->SetDefaultParameter(param_prefix + ":ecalSumRadius", cfg.ecalSumRadius);
+    app->SetDefaultParameter(param_prefix + ":hcalSumRadius", cfg.hcalSumRadius);
+    app->SetDefaultParameter(param_prefix + ":ecalFracSub",   cfg.ecalFracSub);
+    app->SetDefaultParameter(param_prefix + ":hcalFracSub",   cfg.hcalFracSub);
+
+    // initialize particle flow algorithm
+    m_pf_algo.applyConfig(cfg);
+    m_pf_algo.init(logger());
+
+    // initialize geometry service
+    m_geoSvc = app -> template GetService<DD4hep_service>();
+
+  }  // end 'Init()'
+
+
+
+  void ParticleFlow_factory::BeginRun(const std::shared_ptr<const JEvent> &event) {
+
+    // Nothing to do here
+
+  }  // end 'BeginRun(std::shared_ptr<JEvent&>)'
+
+
+
+  void ParticleFlow_factory::Process(const std::shared_ptr<const JEvent> &event) {
+
+    // for calo inputs
+    std::vector<const edm4eic::ClusterCollection*> vecECalInput(m_const.nCaloPairs);
+    std::vector<const edm4eic::ClusterCollection*> vecHCalInput(m_const.nCaloPairs);
+
+    // grab and organize input collections
+    TrkInput trkInput;
+    for (const std::string& input_tag : GetInputTags()) {
+
+      // check if input is a track collection
+      if (input_tag.find("ReconstructedChargedParticles") != std::string::npos) {
+        trkInput.first = static_cast<const edm4eic::ReconstructedParticleCollection*>(event->GetCollectionBase(input_tag));
+        m_log->debug("Found input track collection {}", input_tag);
+        continue;
+      }
+
+      // check if input is a track projection collection
+      if (input_tag.find("TrackProjection") != std::string::npos) {
+        trkInput.second = static_cast<const edm4eic::TrackSegmentCollection*>(event->GetCollectionBase(input_tag));
+        m_log->debug("Found input track projection collection {}", input_tag);
+        continue;
+      }
+
+      // check if input is a calo collection
+      bool isECalCollect = (input_tag.find("Ecal") != std::string::npos);
+      bool isHCalCollect = ((input_tag.find("Hcal") != std::string::npos) || (input_tag.find("HCAL") != std::string::npos));
+      bool endsInClust   = (input_tag.find("Clusters") != std::string::npos);
```

Comment:
Use the types of the collection, not the names (you already realize this since you have had to duplicate the Hcal and HCAL check because people can't and shouldn't be trusted to stick to a single name).